datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2330,4 +2335,22 @@ mod tests {
         assert_eq!(1, schemas.len());
         assert_eq!(0, schemas[0].fields().len());
     }
+
+    #[test]
+    fn test_replace_invalid_placeholder_empty() {
+        let schema = Schema::new(vec![Field::new("id", DataType::Int32, false)]);
+
+        let plan = table_scan(TableReference::none(), &schema, None)
+            .unwrap()
+            .filter(col("id").eq(Expr::Placeholder {
+                id: "".into(),
```

Comment:
https://github.com/apache/arrow-datafusion/issues/5856 also reports an issue if `"$0"` is provided in addition to `""`. Can you please add a test for that as well?